src/utils.ts

```diff
@@ -1301,18 +1301,18 @@ export async function once<T>(ee: EventEmitter, name: string): Promise<T> {
   }
 }
 
-export function maybeAddIdToDocuments(
-  coll: Collection,
+export function maybeAddIdToDocuments<T extends Document>(
```

Comment:
Is this related to the work in this PR or just a nice improvement?

Reply:
Accidentally left over from the PoC, we may want this at some point, but we can do it when necessary.